tests/unit/Metadata/MetadataSerializerTest.php

```diff
@@ -90,4 +91,12 @@ public function deserializes_serialized_data_to_same_representation(): void
 
         self::assertSame('bar', $deserializedMetadata->get('foo'));
     }
+
+    #[Test]
+    public function deserializing_throws_if_contains_invalid_data(): void
```

Comment:
can we have a test for non serializable too?

Reply:
I've added a test for serialization, but perhaps this simply highlights a lack of input validation as the serialization failure cannot happen with valid metadata. Let me know what you think, can improve input validation instead and remove the serialization exception entirely.

Reply:
I believe it would make perfect sense to guard what Metadata accepts, so it can be always serialized, don't think it's much effort?

Reply:
I've moved all input related changes to https://github.com/Lendable/message/pull/77 rather than mix concerns in this PR